handler_test.go

```diff
@@ -127,3 +129,31 @@ func TestStreamRequestRecorder(t *testing.T) {
 		assert.Equal(t, requests, gotRequests)
 	})
 }
+
+func TestIntervalWaiter(t *testing.T) {
+	duration := time.Millisecond * 100
+
+	t.Run("initial call must not wait", func(t *testing.T) {
+		waiter := newIntervalWaiter(duration)
+		now := time.Now()
+		waiter.wait(context.Background())
+		assert.True(t, time.Now().Sub(now) < time.Millisecond)
+	})
+
+	t.Run("interval gap must be 100ms between consecutive calls", func(t *testing.T) {
+		waiter := newIntervalWaiter(duration)
+		waiter.wait(context.Background())
+
+		now := time.Now()
+		waiter.wait(context.Background())
+		duration := time.Now().Sub(now)
+		assert.True(t, duration > (duration), "Unexpected time taken on wait: %v", duration)
+		assert.True(t, duration < (duration), "Unexpected time taken on wait: %v", duration)
+
+		now = time.Now()
+		waiter.wait(context.Background())
+		duration = time.Now().Sub(now)
+		assert.True(t, duration > (duration), "Unexpected time taken on wait: %v", duration)
+		assert.True(t, duration < (duration), "Unexpected time taken on wait: %v", duration)
```

Comment:
should this be in a loop?

Reply:
Yup yup, this was recent commit mistake. And I agree this makes more sense under a loop